[WIP] Create GUI

Prakhar314 opened the pull request:
- [x] Frame the website using HTML and Flask
- [x] Connect python modules. Finish basic functionality.
- [x] Work on styling.

ritwik12 commented:
@Prakhar314 Looks good, Please also add how to use the Web app. Better add a section in Readme about telling the working of this web app so others can contribute to this web app also. Feel free to add credits for the web app in the readme to yourself.

ritwik12 commented:
@Prakhar314 Being Dynamic webpage, will it be possible to host it somewhere? I think gitpages only supports static pages for now.

ritwik12 commented:
@Prakhar314 Added WIP in title for now, please remove that once you are done and let me know so that I can review and merge this.

Prakhar314 commented:
> Being Dynamic webpage, will it be possible to host it somewhere? I think gitpages only supports static pages for now.

Heroku could be used I think.

Prakhar314 commented:
> Added WIP in title for now, please remove that once you are done and let me know so that I can review and merge this.

I think I'm done, please review.

ritwik12 commented:
@Prakhar314 Thanks, let me Review.
Any idea about using Heroku?

ritwik12 commented on app/views.py:
```diff
@@ -0,0 +1,107 @@
+    image_url = StringField(
+        "image_url",
+        validators=[validators.Optional(), validators.URL()],
+        render_kw={"placeholder": "Enter a URL"},
```
@Prakhar314 Fine with `Enter a URL` too, but any other suggestions to make it more cool?

ritwik12 commented:
@Prakhar314 Appreciate your hard work here, Amazing contribution.

ritwik12 commented:
@Prakhar314 I can see that you have not made any changes to a lot of files but still they are showing in the Diff for your PR. Can you please fix that?

ritwik12 commented:
@Prakhar314 You have done an amazing job here putting up the GUI and I think with the GUI a more people will be able to use and contribute to the project.

I am really interested in hosting the Web App too. Will you be able to work on that too?

Prakhar314 commented:
> @Prakhar314 I can see that you have not made any changes to a lot of files but still they are showing in the Diff for your PR. Can you please fix that?

I can't figure out how to do that. Do you know how to?
All I could find was how to hide them temporarily([link](https://cloudfour.com/thinks/quick-tip-how-to-hide-whitespace-changes-in-git-diffs/))

Prakhar314 commented on app/templates/about.html:
```diff
@@ -0,0 +1,71 @@
+</div>
+{% endblock %}
+{% block script %}
+<script>
+	// send request for project description
+	console.log("yp");
+	var request = new XMLHttpRequest();
+	request.open('GET', 'https://api.github.com/repos/ritwik12/Celestial-bodies-detection');
+	request.send();
+	// set description
+	request.onload = async function () {
+		var data = JSON.parse(this.response);
+		console.log(data.description);
+		document.getElementById('aboutProject').innerHTML = data.description;
+	}
```
I guess we can. What do you want to be shown?
I was showing the list of contributors, but commented it out for now (the above review).

ritwik12 replied:
@prakhar214 We can show contributors as well, but we should show something about the project in detail. You can use anything from Readme.md or summarise based on the understanding. Basically it should give an idea to the people coming to the website about the Project and what it does.

ritwik12 commented:
@Prakhar314 so when you are doing `git add` in your git repository. Make sure you only add the files you have worked upon and not do `git add .`. After that you can commit normally and it will show like usual. for now I think you have directly committed to the Master branch and have not created a new branch.

Whenever you are working on something which is not a small contribution then always create a new branch and push changes to that only. This helps in cases like these to revert things and fix them.

For now if you want, you will have to store your changes to some other places, then hard reset your master branch which will make it back to head of the upstream. Then make the changes and pushing them.

Be aware that when you reset the master it will not have any diff which eventually will close this PR 😅. so for now I think you can go like this only. There are no issues.

ritwik12 commented:
I think https://cloudfour.com/thinks/quick-tip-how-to-hide-whitespace-changes-in-git-diffs/ is for the reviewer only while reviewing to hide those. You can also hide it but for your VIEW only not for others. And Hiding it does not make it get away from there. They are there.

ritwik12 commented:
@Prakhar314 Please check the reviews and let me know if you have any doubt. Hope you are not in a hurry, if you have worked hard I want it to be perfect at last.

ritwik12 commented:
@Prakhar314 Any updates here?

Prakhar314 commented:
I rebased and force pushed to remove whitespace changes. I also changed some of the button names as asked.
I won't be getting time to update this for a while.
I'll notify when I make changes.

ritwik12 commented:
@Prakhar314 Ohk sure, Thanks :)

ritwik12 commented:
@Prakhar314 Did you got time to update? Let's just add screenshots to Readme and Merge this one.

ritwik12 commented:
@Prakhar314 Reminder!!

ritwik12 commented:
@Prakhar314 Final nudge, Lets just resolve conflicts to this PR and merge it. we can work on improvements later.

ritwik12 commented:
@Prakhar314 Merged this, thanks!!
